### PR TITLE
feat: add vatsim command

### PR DESCRIPTION
**Pull request by pdellaert**
feat: add vatsim command to display VATSIM online facilities based on user search

## Description
This command will provide a list of the VATSIM facilities and ATIS that are currently online with some basic information for each. There is a limit of 5 maximum.

## Test Results
Screen Shot 2022-08-08 at 11 38 26 PM
Screen Shot 2022-08-08 at 11 43 59 PM
Screen Shot 2022-08-08 at 11 45 03 PM
Screen Shot 2022-08-08 at 11 45 18 PM

## Discord Username

straks#7240

### CODE REVIEW

File: .github/command-docs.md [hunk 1]
```diff
@@ -95,21 +95,22 @@
...
+| .liveflights | Get the current live flights for FlyByWire Simulations                  | .flights <br/> .flight                                             |
+| .membercount | Lists the guild's current amount of members                             | ---                                                                |
+| .metar       | Provides the METAR report of the requested airport                      | ---                                                                |
+| .ping        | Send back a message                                                     | ---                                                                |
+| .roleinfo    | Lists the guild's current amount of members                             | ---                                                                |
+| .station     | Provides station information                                            | ---                                                                |
+| .vatsim      | Show a list of Online VATSIM facilities based on a user provided search | .vatsimfacilities <br/> .vf                                        |
```
Comment (benw202) [inline .github/command-docs.md:111]:
```suggestion
| .vatsim      | Show a list of online VATSIM facilities based on a user provided search | .vatsimfacilities <br/> .vf                                        |
```

Should this be lower case?

**Comment by benw202** [comment]
(Drafted and undrafted to see if new check would run)

**Comment by benw202** [comment]
Staging has been merged in to make use of new check. 

I have bumped the date forwards as the changelog entry for Vals PR is the 10th as he is ahead and this will be merged 10th or later haha!

**Comment by NathanInnes** [comment]
I have a proposal to also add vatsim pilot information into this command also, so when a user types a callsign it returns the pilot's information as well.

**Comment by pdellaert** [comment]
Did a whole redesign/refactor to include much more functionality and logic. I'm pretty sure the code is not great quality, will look into that later. Happy to receive any reviews/feedback. 

Check out screenshots for all the details:
Screen Shot 2022-08-10 at 12 43 53 AM
Screen Shot 2022-08-10 at 12 45 24 AM
Screen Shot 2022-08-10 at 12 45 41 AM
Screen Shot 2022-08-10 at 12 47 05 AM
Screen Shot 2022-08-10 at 12 47 17 AM
Screen Shot 2022-08-10 at 12 47 38 AM
Screen Shot 2022-08-10 at 12 47 51 AM
Screen Shot 2022-08-10 at 12 48 03 AM
Screen Shot 2022-08-10 at 12 48 19 AM

File: src/commands/utils/vatsimData.ts [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+            commandMode = 'STATS';
+        } else if (subcommandTerms.startsWith(' controllers')) {
+            searchTerms = subcommandTerms.replace(/\scontrollers\s+/, ' ').split(' ');
+            commandMode = 'CONTROLLERS';
+            notFoundMsg = 'No online VATSIM Controllers, Observers or ATIS found matching your search query.';
+        } else if (subcommandTerms.startsWith(' pilots')) {
+            searchTerms = subcommandTerms.replace(/\spilots\s+/, ' ').split(' ');
+            commandMode = 'PILOTS';
+            notFoundMsg = 'No online VATSIM Pilots found matching your search query.';
+        }
```
Comment (NathanInnes) [inline src/commands/utils/vatsimData.ts:73]: Switch case statement may be a cleaner solution

File: src/commands/utils/vatsimData.ts [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+
+        try {
+            const vatsimData = await fetch(DATA_VATSIM_URL)
+                .then((res) => res.json());
+
+            if (commandMode === 'STATS') {
+                const fields: EmbedField[] = [
+                    {
+                        name: 'Pilots',
+                        value: vatsimData.pilots ? vatsimData.pilots.length : 0,
```
Comment (NathanInnes) [inline src/commands/utils/vatsimData.ts:83]: Recommend splitting these into const naming them pilotCount or something to make it more clear what the value is

Comment (NathanInnes) [reply src/commands/utils/vatsimData.ts:83]: Like you did on line 106

File: src/commands/utils/vatsimData.ts [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+                }).slice(0, 5).flat();
+
+                const totalCount = Object.keys(vatsimControllers).length + Object.keys(vatsimAtis).length;
+                const shownCount = totalCount <= 5 ? totalCount : 5;
+                responseEmbed = makeEmbed({
+                    title: `VATSIM Data | ${searchTerm} - ${totalCount} Controllers, Observers & ATIS online`,
+                    description: `A list of ${shownCount} online VATSIM Controllers, Observers & ATIS matching ${searchTerm}`,
+                    fields,
+                });
+            } else if (Object.keys(vatsimPilots).length !== 0 && (commandMode === 'ALL' || commandMode === 'PILOTS')) {
```
Comment (NathanInnes) [inline src/commands/utils/vatsimData.ts:166]: This is being a bit tedious but this is being used more than once, maybe make it a const so you can reuse it?
```ts
Object.keys(vatsimPilots).length
```

File: src/commands/utils/vatsimData.ts [hunk 5]
```diff
@@ -0,0 +1,233 @@
...
+                const fields: EmbedField[] = [...vatsimPilots.sort((a, b) => b.pilot_rating - a.pilot_rating)].map((vatsimItem) => {
+                    // eslint-disable-next-line camelcase
+                    const { callsign, pilot_rating, flight_plan } = vatsimItem;
+                    // eslint-disable-next-line camelcase
+                    const ratingDetail = vatsimPilotRatings.filter((ratingInfo) => ratingInfo.id === pilot_rating);
+                    const ratingText = `${ratingDetail[0].short_name} - ${ratingDetail[0].long_name}`;
```
Comment (NathanInnes) [inline src/commands/utils/vatsimData.ts:172]: Prefer array destructuring

**Review by NathanInnes — COMMENTED** [review]
Very nice command, see review details

**Comment by pdellaert** [comment]
Did another code restructure based on what I learned from the Temporary Command stuff. Did make sure to cover the above review comments fully, I hope. 

### Test evidence
Screen Shot 2022-08-14 at 11 59 20 PM
Screen Shot 2022-08-14 at 11 59 34 PM
Screen Shot 2022-08-15 at 12 00 12 AM
Screen Shot 2022-08-15 at 12 00 53 AM
Screen Shot 2022-08-15 at 12 01 02 AM
Screen Shot 2022-08-15 at 12 01 13 AM
Screen Shot 2022-08-15 at 12 01 41 AM
Screen Shot 2022-08-15 at 12 02 11 AM
Screen Shot 2022-08-15 at 12 02 58 AM
Screen Shot 2022-08-15 at 12 03 26 AM
Screen Shot 2022-08-15 at 12 13 32 AM
Screen Shot 2022-08-15 at 12 20 02 AM

**Review by NathanInnes — APPROVED** [review]

**Review by benw202 — APPROVED** [review]

**Pull request merged**